src/Bridge/Laravel/Console/SamConfigureRoutes.php

```diff
@@ -28,33 +29,53 @@ class SamConfigureRoutes extends Command
      */
     protected $description = 'Configure the SAM Template with our Laravel Routes';
 
-    public function handle(): int {
+    public function handle(): int
+    {
         $samConfig = Yaml::parseFile(base_path('template.yaml'), Yaml::PARSE_CUSTOM_TAGS);
-
-        $samConfig['Resources']['Website']['Properties']['FunctionName'] = strtolower(env('APP_NAME').'-apigateway');
+        $samConfig['Resources']['Website']['Properties']['FunctionName'] = strtolower(env('APP_NAME') . '-apigateway');
 
         // Handle the website events.
         $samConfig['Resources']['Website']['Properties']['Events'] = [];
         /** @var RouteCollection $routeCollection */
         $routeCollection = Route::getRoutes();
         /** @var \Illuminate\Routing\Route $route */
-        foreach($routeCollection->getRoutes() as $route){
-            foreach($route->methods() as $method)
-                $routeName = ($route->uri() == '/') ? 'root' : preg_replace('/[^A-Za-z0-9\-]/', '', $route->uri());
-                $name = $route->getName() ?: sprintf("%s%s", ucfirst(strtolower($method)), ucfirst(strtolower($routeName)));
-                $method = strtoupper($method);
-                $path = $route->uri()[0] == '/' ? $route->uri() : '/'.$route->uri();
-                $samConfig['Resources']['Website']['Properties']['Events'][] = [
-                    $name => [
-                        'Type' => 'Api',
-                        'Properties' => [
-                            "Path" => $path,
-                            "Method" => $method
-                        ]
-                    ]
-                ];
-            }
+        foreach ($routeCollection->getRoutes() as $route) {
+            $methods = $route->methods();
+            (collect($methods))->each(function (string $method) use ($route, &$samConfig) {
+                list($name, $config) = $this->routing($method, $route->uri, $route->getName());
+                $this->info('Setting: ' . $name);
+                $samConfig['Resources']['Website']['Properties']['Events'][$name] = $config;
```

Comment:
Yes, that is what drove me to it. But then there is also the fact that by not having a catch-all route, we don't have to pay Lambda costs for bots that are attempting to search through URLs that are no good, because the API Gateway will simply block those for us.

I believe that we can make it even more mature over time. But for now, it seemed the best solution.

Reply:
I was tempted to do something similar, and I like your rationalisation about the costs for non-existent routes! However, does that mean that your application won't be able to supply custom 404 pages to the user?

Reply:
The short answer is yes, that is what it means. The longer answer is that we can customize API Gateway error responses. 

Or, we could simply change the route configuration to always add the catch-all route as well as defined routes. Then, when running on AWS, the catch-all will do it. I think longer term, I would personally prefer to limit the vulnerability by having API Gateway block those.

What do you reckon?

Reply:
I think it depends on the application you're running on Lambda, and how important the 404 entry in the application log is to you / your users.

If I'm serving a REST API, for example, then I don't care what the user sees (as long as the `404` status code is correct), and those logs probably aren't that important compared to other entries in the application log. In this case, I'd also definitely appreciate API Gateway rejecting 'bad' traffic.

Serving a standard website, however, is a different matter. I want to show the user a nice 404 page with links to help them find their way, and I'll probably want to have the information go to Loggly or something so I can detect and diagnose broken routes in my system.

Perhaps this needs to be an optional configuration setting the user can choose?

(I have to admit, I haven't done custom API Gateway error responses, so maybe my argument here is invalid)

Reply:
I will just add the catchall as a default route for now. We can refine it later.

Reply:
@nealio82 If you want to jump on the Bref Slack to discuss things ... https://join.slack.com/t/brefworkspace/shared_invite/enQtNTQyNTQ2NDM4NTgzLTBkYTYyNGEyYWIxNzE4NmM5N2QyM2E1OTI3NTM0NDdhZDE2YzI2MGZjNDRmNDMzNzBmMmJiZjVlY2Q1Yjg4NDg come on!